client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -305,6 +305,22 @@ Undocumented.prototype.getJetpackConnectionStatus = function( siteId, fn ) {
 	}, fn );
 };
 
+/*
+ * Retrieve current user's connection data for a Jetpack site.
+ *
+ * @param {int}      [siteId]
+ * @param {Function} fn
+ * @api public
+ */
+Undocumented.prototype.getJetpackUserConnectionData = function( siteId, fn ) {
```

Comment:
Yes, that's a good point. I intentionally didn't use the data layer though, because I strived for consistency with the rest of the methods that communicate with the Jetpack site REST API. 

It would be a good improvement, of course, but I'd prefer to leave it for another PR (or set of PRs) that would migrate all of these actions to use the data layer.